src/mutation.cpp

```diff
@@ -835,28 +835,30 @@ void player::mutate_towards(std::string mut)
         mutation_effect(*this, mut);
         mutation_replaced = true;
     }
-    if (canceltrait != "") {
-        if(traits[mut].mixed_effect || traits[canceltrait].mixed_effect) {
-            rating = m_mixed;
-        } else if(traits[mut].points <= 0 && traits[canceltrait].points > 0) {
-            rating = m_good;
-        } else if(traits[mut].points > 0 && traits[canceltrait].points <= 0) {
-            rating = m_bad;
-        } else if(traits[mut].points == 0 && traits[canceltrait].points == 0) {
-            rating = m_neutral;
-        } else {
-            rating = m_mixed;
+    if (canceltrait.size() > 0) {
+        for (size_t i = 0; i < canceltrait.size(); i++) {
+            if(traits[mut].mixed_effect || traits[canceltrait[i]].mixed_effect) {
+                rating = m_mixed;
+            } else if(traits[mut].points <= 0 && traits[canceltrait[i]].points > 0) {
+                rating = m_good;
+            } else if(traits[mut].points > 0 && traits[canceltrait[i]].points <= 0) {
+                rating = m_bad;
+            } else if(traits[mut].points == 0 && traits[canceltrait[i]].points == 0) {
+                rating = m_neutral;
+            } else {
+                rating = m_mixed;
```

Comment:
Also: `if(traits[mut].points <= 0 && traits[canceltrait[i]].points > 0) { rating = m_good; }` isn't `traits[mut]` the mutation we're currently getting, and `traits[canceltrait[i]]` the one we're cancelling? So if the new trait has less than zero points and the old one had more than zero, this is a good thing? Kind of contradicting what you just said there.